### ISSUE TITLE
False positive in if statement

### ISSUE THREAD

**Issue by lnader** [SonarSource/SonarJS#911]
**I want to report a bug.**

**SonarJS version:** 1.0.0<br>

**Reproducer** 
```
const x = {foo:4, bar:"abc"}
  if (x && x.bar) {
    return x;
  }
```

MAJOR - S2589: /***/test.js [410, 7]: Refactor this code so that this expression does not always evaluate to true.

**Comment by vilchik-elena** [comment]
@lnader what's wrong? "x" is useless in your condition
(and you probably put the wrong version of SonarJS)

**Comment by lnader** [comment]
@vilchik-elena 
This is the version of SonarJs returned from command
$ sonarjs -v
1.0.0

Why is x useless?  I was trying to give a simple reproducible code snippet but the actual code could be

```
const x = makeHttpRequest(foo, bar)
//if error return error
if (x && x.err) {
    return x;
}
//else process results
.....
```

**Comment by vilchik-elena** [comment]
Ah, sorry, you are using CLI.

This is symbolic execution rule, I need more context to understand the problem. Could you provide full function? (and `makeHttpRequest` as well)
